src/main/java/org/truffleruby/language/control/CheckIfPatternsMatchedNode.java

```diff
@@ -17,14 +18,16 @@
 public final class CheckIfPatternsMatchedNode extends RubyContextSourceNode {
 
     @Child RubyNode inspected;
+    @Child ToJavaStringNode toJavaStringNode;
 
     public CheckIfPatternsMatchedNode(RubyNode inspected) {
         this.inspected = inspected;
     }
 
     @Override
     public Object execute(VirtualFrame frame) {
-        String message = inspected.execute(frame).toString();
+        toJavaStringNode = ToJavaStringNode.create(inspected);
+        String message = (String) toJavaStringNode.execute(frame);
```

Comment:
Use `String message = toJavaStringNode.executeToJavaString(inspected.execute(frame));` and then no need to cast